### PR TITLE
SYCL. Add dask support for distributed calculations with sycl

### PR DESCRIPTION
**Pull request by razdoburdin**
Hi,
I continue adding of sycl support (https://github.com/dmlc/xgboost/pull/10806, https://github.com/dmlc/xgboost/pull/10701, etc).
This PR adds support for distributed calculations with sycl and add one simple test for basic verification.

### CODE REVIEW

File: python-package/xgboost/core.py [hunk 1]
```diff
@@ -306,11 +306,12 @@ def _check_distributed_params(kwargs: Dict[str, Any]) -> None:
         raise TypeError(msg)
 
     if device and device.find(":") != -1:
-        raise ValueError(
-            "Distributed training doesn't support selecting device ordinal as GPUs are"
-            " managed by the distributed frameworks. use `device=cuda` or `device=gpu`"
-            " instead."
-        )
+        if device != "sycl:gpu":
```
Comment (trivialfis) [inline python-package/xgboost/core.py:309]: What happens if it's sycl GPU, is dask going to select the appropriate GPU?

File: plugin/sycl/tree/hist_updater.cc [hunk 2]
```diff
@@ -22,6 +22,32 @@ using ::sycl::ext::oneapi::plus;
...
+template <typename GradientSumT>
+void HistUpdater<GradientSumT>::ReduceHists(const std::vector<int>& sync_ids,
+                                            size_t nbins) {
+  if (reduce_buffer_.size() < sync_ids.size() * nbins) {
+    reduce_buffer_.resize(sync_ids.size() * nbins);
+  }
+  for (size_t i = 0; i < sync_ids.size(); i++) {
+    auto& this_hist = hist_[sync_ids[i]];
+    const GradientPairT* psrc = reinterpret_cast<const GradientPairT*>(this_hist.DataConst());
+    // std::copy(psrc, psrc + nbins, reduce_buffer.begin() + i * nbins);
```
Comment (trivialfis) [inline plugin/sycl/tree/hist_updater.cc:34]: Please remove the commented code.

File: python-package/xgboost/core.py (hunk 1 above)
Comment (razdoburdin) [reply python-package/xgboost/core.py:309]: Yes, the appropriate GPUs would be selected

File: plugin/sycl/tree/hist_updater.cc (hunk 2 above)
Comment (razdoburdin) [reply plugin/sycl/tree/hist_updater.cc]: done

**Comment by razdoburdin** [comment]
Hi @trivialfis ,
what is your opinion about this PR?

File: python-package/xgboost/core.py (hunk 1 above)
Comment (trivialfis) [reply python-package/xgboost/core.py:309]: Got it.

**Review by trivialfis — APPROVED** [review]

**Pull request merged**